src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -313,7 +296,17 @@ public sealed override bool Equals(object obj)
         {
             var t2 = obj as TypeSymbol;
             if ((object)t2 == null) return false;
-            return this.Equals(t2, TypeCompareKind.ConsiderEverything);
+            return this.Equals(t2, SymbolEqualityComparer.Default);
+        }
+
+        bool ITypeComparable.Equals(ISymbol other, TypeCompareKind compareKind)
```

Comment:
>ITypeComparable.Equals [](start = 13, length = 22)

Should `ITypeComparable` be implemented once, on `Symbol`, with the implementation of `Equals(ISymbol other, TypeCompareKind compareKind)` in `Symbol` delegating to a `virtual bool Equals(Symbol other, TypeCompareKind compareKind)`?

Reply:
That's an interesting idea, I think that'd work nicely. The current design is from before we had it on ISymbol, but now we do should make it clearer. I'll try it out.

---
In reply to: [305470115](https://github.com/dotnet/roslyn/pull/37247#discussion_r305470115) [](ancestors = 305470115)